src/app/services/showcases/showcases.service.ts

```diff
@@ -1,33 +1,29 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
-import {  BehaviorSubject } from 'rxjs';
+import { BehaviorSubject, EMPTY, throwError, Observable } from 'rxjs';
 import { ShowcaseItem } from 'src/app/data/showcase-item';
-
+import { tap, catchError } from "rxjs/operators";
 
 @Injectable({
   providedIn: 'root'
 })
 export class ShowcasesService {
 
-  public showCasesSubject:BehaviorSubject<ShowcaseItem[]>=new BehaviorSubject<ShowcaseItem[]>(null);
-  SERVER_URL: string = "http://localhost:8080/api/";
-  constructor(private httpClient:HttpClient) {
-    this.getAllShowCases()
-  }
+  SERVER_URL: string = "api/showcases";
 
 
-  private getAllShowCases()
-  {
+  constructor(private httpClient: HttpClient) {
+  }
 
-    let data;
-    this.httpClient.get(this.SERVER_URL + 'showcases').subscribe(response=>
-    {
-      data=response;
-      this.showCasesSubject.next(data);
-      setTimeout(() => {
-        this.getAllShowCases()
-    }, 5000);
-    })
 
+  getAllShowCases(): Observable<ShowcaseItem[]> {
```

Comment:
You can create a behavior subject with an initial empty array then store the showcases in, Easier to share the data that way.

Not entirely sure if it's best practice for that case, will let @Yejielw decide.

An example would be 
```ts
private showcasesSource = new BehaviorSubject([]);
showcases$ = this.showcasesSource.asObservable(); // Creating a branch observable so we can't directly use next() from the behavior subject. This is what you'll use in components.

private setShowcases(data) {
  return this.showcasesSource.next(data)
}

// ... Tap into the HTTP get request and set the response to showcasesSource subject...
```